app/react/V2/Routes/Settings/Filters/FiltersTable.tsx

```diff
@@ -70,43 +71,57 @@ const FiltersTable = () => {
   }, [blocker, setConfirmNavigationModal]);
 
   const cancel = () => {
-    setFilters(loaderData.filters);
+    setFilters(loadedFilters);
   };
 
   const addNewFilters = (templatedIds: string[]) => {
     const newFilters = createNewFilters(templatedIds, templates);
-    setFilters([...(filters || []), ...newFilters]);
+    setFilters([...filters, ...newFilters]);
   };
 
   const handleDelete = () => {
-    const idsToRemove = selectedFilters.map(selected => selected.original.id);
+    const idsToRemove: string[] = [];
+    filters?.forEach(filter => {
+      if (filter.rowId in selectedFilters) {
+        idsToRemove.push(filter.rowId);
+      }
+      if (filter.subRows) {
+        filter.subRows.forEach(subRow => {
+          if (subRow.rowId in selectedFilters) {
+            idsToRemove.push(subRow.rowId);
+          }
+        });
+      }
+    });
+
     const updatedFilters = deleteFilters(filters, idsToRemove);
-    setFilters(updatedFilters);
+    setFilters(updatedFilters || []);
   };
 
   const handleSave = async () => {
     setDisabled(true);
-    const response = await settingsAPI.save({ filters: sanitizeFilters(filters) });
-
+    const savedFilters = sanitizeFilters(filters);
+    const response = await settingsAPI.save({ filters: savedFilters });
     if (response instanceof FetchResponseError) {
       return setNotifications({
         type: 'error',
         text: <Translate>An error occurred</Translate>,
         ...(response.message && { details: response.message }),
       });
     }
-
     setSettings(response);
     setDisabled(false);
     setHasChanges(false);
+    const formattedFilters = formatFilters(savedFilters || []);
```

Comment:
Could it be that the reported inconsistencies are related with keeping the local filters instead of the data in the response?

Reply:
Kinda yes. It was not a good idea to avoid using the revalidator.